atom/browser/web_contents_preferences.cc

```diff
@@ -147,6 +144,18 @@ WebContentsPreferences::WebContentsPreferences(
 #endif
   SetDefaultBoolIfUndefined(options::kOffscreen, false);
 
+  if (Browser::Get()->secure_mode_enabled()) {
```

Comment:
This might have gotten lost in the original comment but I meant that we should always do the `SetDefaultBoolIfUndefined` calls for the normal defaults.  And if `secure_mode_enabled()` then override the values regardless of whether they are undefined.  E.g.

```cpp
SetBoolIfUndefined(optionA, true);
if (is_secure_mode) {
  SetBool(optionA, false);
}
```

Then we can remove the `if (!secure_mode_enabled && IsEnabled(options::kNodeIntegration))` style code below (we won't need to check `secure_mode_enabled`)

Reply:
some options need to stay overridable. I only want to kill `nodeIntegration` and `nodeIntegrationInWorker` as those are not necessary to keep functionality as you can use preload script instead.

Reply:
or maybe it should be configurable, which options are going to be disabled completely.

Reply:
something like?
```
interface OptionsToDisable {
  sandbox?: boolean;
  contextIsolation?: boolean;
  nativeWindowOpen?: boolean;
  enableRemoteModule?: boolean;
  nodeIntegration?: boolean;
  nodeIntegrationInWorker?: boolean;
}

app.enableSecureMode(optionsToDisable: OptionsToDisable): void;
```

Reply:
options will always default to the more secure value and if specified in this list, they will also not be overridable.

Reply:
e.g.
```ts
app.enableSecureMode({
    contextIsolation: true // allow disabling contextIsolation
});
```

Reply:
or some other variant of this, to make the API as least confusing as possible

Reply:
> some options need to stay overridable.

I disagree, if the options can be overridable then any of the previous vulns we've had where people have been able to set `nodeIntegration: true` (somehow) would still work making this option not very helpful.

I think this option should give no options and violently enforce the security guidelines (disable node, enable isolation, reject permission requests if not already handled, etc..)

That's just IMO though, would like some weigh in from @electron/wg-security

Reply:
Well, `contextIsolation` for example is not an option for us for now (we'll need to do a lot of refactoring to get there). We need it to be optionally overridable. The default can be that nothing is overridable and you have to opt-in explicitly into being able to override some options.